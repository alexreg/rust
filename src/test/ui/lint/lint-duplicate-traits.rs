// compile-pass

#![feature(trait_alias)]

trait Foo {}

trait SyncAlias = Sync;

impl Foo for dyn Send {}

impl Foo for dyn Send + Send {}

impl Foo for dyn Send + Sync + Send + SyncAlias {}

fn main() {}
