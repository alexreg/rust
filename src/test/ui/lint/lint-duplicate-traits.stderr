warning: duplicate auto trait `std::marker::Send` found in trait object
  --> $DIR/lint-duplicate-traits.rs:13:18
   |
LL | impl Bar for dyn Send + Send {}
   |                  ^^^^   ^^^^ subsequent use of auto trait
   |                  |
   |                  first use of auto trait
   |
   = note: #[warn(duplicate_auto_traits_in_trait_objects)] on by default

warning: duplicate auto trait `std::marker::Send` found in trait object
  --> $DIR/lint-duplicate-traits.rs:15:18
   |
LL | impl Baz for dyn Send + Sync + Send + SyncAlias {}
   |                  ^^^^          ^^^^ subsequent use of auto trait
   |                  |
   |                  first use of auto trait

warning: duplicate auto trait `std::marker::Sync` found in trait object
  --> $DIR/lint-duplicate-traits.rs:15:25
   |
LL | trait SyncAlias = Sync;
   |                   ---- subsequent use of auto trait
...
LL | impl Baz for dyn Send + Sync + Send + SyncAlias {}
   |                         ^^^^          ^^^^^^^^^
   |                         |
   |                         first use of auto trait

warning: duplicate auto trait `std::marker::Send` found in trait object
  --> $DIR/lint-duplicate-traits.rs:13:18
   |
LL | impl Bar for dyn Send + Send {}
   |                  ^^^^   ^^^^ subsequent use of auto trait
   |                  |
   |                  first use of auto trait

